### PR TITLE
EOS-20503: Disable 2GB swap on LR R1 SMS2

### PR DESCRIPTION
**Pull request by SomwanshiAnjali**
Signed-off-by: Anjali Somwanshi <anjali.somwanshi@seagate.com>

### CODE REVIEW

**Review by ypise — APPROVED** [review]

File: srv/components/system/storage/config.sls [hunk 1]
```diff
@@ -18,10 +18,16 @@
...
-# Make SWAP
-Ensure SWAP partition is unmounted:
-  cmd.run:
-    - name: swapoff -a && sleep 2
+# Disable system swap
+Deactivate and unmount system SWAP:
+  module.run:
+    # Deactivate SWAP
+    - mount.swapoff:
+      - name: /dev/dm-1
```
Comment (83bhp) [inline srv/components/system/storage/config.sls:26]: Can we not use /dev/mapper/vg_sysvol-lv_swap instead /dev/dm-1?

**Review by 83bhp — APPROVED** [review]
Minor comment, take care if it's valid. Everything else looks good to me.

File: srv/components/system/storage/config.sls (hunk 1 above)
Comment (ypise) [reply srv/components/system/storage/config.sls:26]: @83bhp 
SWAP unfortunately prefers `/dev/dm-1` for some reason for `swapoff`:
https://github.com/Seagate/cortx-prvsnr/blob/pre-cortx-1.0/srv/components/system/storage/config/config.sls

For `swapon` what you have suggested works and can be incorporated in teardown, if it helps.

Comment (83bhp) [reply srv/components/system/storage/config.sls:26]: Oh, ok.

**Comment by 83bhp** [comment]
Please test the patch before merging.

File: srv/components/system/storage/teardown/init.sls [hunk 2]
```diff
@@ -22,8 +22,13 @@ include:
 {% endif %}
 
 Mount default OS swap device:
-  cmd.run:
-    - name: swapon -a || true
+  module.run:
+    - mount.swapon:
+      - name: /dev/dm-1
```
Comment (ypise) [inline srv/components/system/storage/teardown/init.sls:27]: @83bhp is correct.
This is unreliable.
```
[root@srvnode-1 ~]# swapon
NAME      TYPE       SIZE USED PRIO
/dev/dm-6 partition 12.8G   0B   -2
```
 We need to query and switch-off based on system data.

**Review by ypise — CHANGES_REQUESTED** [review]

File: srv/components/system/storage/teardown/init.sls (hunk 2 above)
Comment (ypise) [reply srv/components/system/storage/teardown/init.sls:27]: @SomwanshiAnjali 
This is a better alternative that works:
```
# Disable system swap
{% set swap_devices = salt['mount.swaps']().keys() %}
Deactivate and unmount system SWAP:
  module.run:
    # Deactivate SWAP
    {% for swap_device in swap_devices %}
    - mount.swapoff:
      - name: {{ swap_device }}
    {% endfor %}
    # Remove SWAP from /etc/fstab
    - mount.rm_fstab:
      - name: swap
      - device: /dev/mapper/vg_sysvol-lv_swap
```

Result:
```
local:
    ----------
    Deactivate and unmount system SWAP:
        ----------
        module.run:
            |_
              ----------
              mount.swapoff:
                  |_
                    ----------
                    name:
                        /dev/dm-6
            |_
              ----------
              mount.rm_fstab:
                  |_
                    ----------
                    name:
                        swap
                  |_
                    ----------
                    device:
                        /dev/mapper/vg_sysvol-lv_swap
```

Comment (SomwanshiAnjali) [reply srv/components/system/storage/teardown/init.sls]: @ypise 
Addressed, please check

**Review by ypise — APPROVED** [review]

**Review by 83bhp — APPROVED** [review]

**Pull request merged**